### ISSUE TITLE
GrpcWebSocketBridge: Buffer is not large enough for header

### ISSUE THREAD

**Issue by Sen-Gupta** [Cysharp/GrpcWebSocketBridge#26]
Hello Team

The issue still occurs.

Environment: .Net 8, MagicOnion 6.0, GrpcWebSocketBridge 6.0
Network: Client->Cloudflare-Firewall->Proxy-Gprc

Observations: We see MagicOnion returning  a success message
Request finished HTTP/2 POST https://www.fixme.com/IPortfoliosGrpcService/GetAllEntitiesAsync - 200 null application/grpc 20.9603ms

The grpc websocket bridge seems to be broken at the hightlighted code below. 

ExceptionDetail.InnerException.Message
Buffer is not large enough for header
ExceptionDetail.InnerException.Source
GrpcWebSocketBridge.Client
ExceptionDetail.InnerException.TargetSite
Void MoveNext()
ExceptionDetail.InnerException.Type
System.InvalidOperationException
ExceptionDetail.Message
**Status(StatusCode="Internal", Detail="Error starting gRPC call. InvalidOperationException: Buffer is not large enough for header",** DebugException="System.InvalidOperationException: Buffer is not large enough for header")
ExceptionDetail.Source
System.Private.CoreLib
ExceptionDetail.Status._typeTag
Status
ExceptionDetail.Status.DebugException._typeTag
InvalidOperationException
ExceptionDetail.Status.DebugException.Data
[]
ExceptionDetail.Status.DebugException.HelpLink
 
ExceptionDetail.Status.DebugException.HResult
-2146233079
ExceptionDetail.Status.DebugException.InnerException
 
ExceptionDetail.Status.DebugException.Message
Buffer is not large enough for header
ExceptionDetail.Status.DebugException.Source
**GrpcWebSocketBridge.Client
ExceptionDetail.Status.DebugException.StackTrace
   at Grpc.Net.Client.Web.Internal.GrpcWebResponseStream.ReadAsync(Memory`1 data, CancellationToken cancellationToken)
   at Grpc.Net.Client.Internal.StreamExtensions.ReadMessageAsync[TResponse](Stream responseStream, GrpcCall call, Func`2** deserializer, String grpcEncoding, Boolean singleMessage, CancellationToken cancellationToken)
   at Grpc.Net.Client.Internal.GrpcCall`2.ReadMessageAsync(Stream responseStream, String grpcEncoding, Boolean singleMessage, CancellationToken cancellationToken)
   at Grpc.Net.Client.Internal.GrpcCall`2.RunCall(HttpRequestMessage request, Nullable`1 timeout)
ExceptionDetail.Status.DebugException.TargetSite
Void MoveNext()
ExceptionDetail.Status.Detail
Error starting gRPC call. InvalidOperationException: Buffer is not large enough for header
ExceptionDetail.Status.StatusCode
Internal
ExceptionDetail.StatusCode
Internal
ExceptionDetail.TargetSite
Void Throw()
ExceptionDetail.Trailers
[]

**Comment by licentia88** [comment]
I have come across this issue too, this exception is thrown in the latest version (version 6)

I've downgraded to the following version  and it works.

MagicOnion.Client 5.1.8
MagicOnion.Shared 5.1.8
MagicOnion.Server 5.1.8
MagicOnion.Server.HttpGateway 5.1.8
MagicOnion.Serialization.MemoryPack 5.1.8-preview

**Comment by KAW0** [comment]
I have also encountered that issue, but the solution is simple.
Change:
```cs
 var channel = GrpcChannel.ForAddress(address, new GrpcChannelOptions()
                {
                    HttpHandler = new GrpcWebSocketBridgeHandler()
                });
```
To:
```cs
 var channel = GrpcChannel.ForAddress(address, new GrpcChannelOptions()
                {
                    HttpHandler = new GrpcWebSocketBridgeHandler(true)
                });
```

**Comment by Sen-Gupta** [comment]
@KAW0, 

Thanks it works as expected with the changes.

**Comment by KAW0** [comment]
While this solve this problem I don't think that issue is fully resolved since this parameter is undocumented.  It easy to just follow basic example from Readme without realizing that this is and why it is needed.

**Comment by Sen-Gupta** [comment]
Actually, it is partial solution.
The setting won't work in MAUI project.

There were more changes required to make it work on Android.

I will document it here in a while.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by mayuki** [comment]
We just released version 1.3.0.
Could you please check if the problem occurs in the new version?

**Comment by mayuki** [comment]
I have forwarded it to a separate Issue because it is a different problem than this one that has been reported.
